frontend/test/pytest/test_loops.py

```diff
@@ -253,6 +254,47 @@ def inner(j):
         assert circuit(5, 6) == 30  # 5 * 6
         assert circuit(4, 7) == 28  # 4 * 7
 
+    @pytest.mark.usefixtures("disable_capture")
+    def test_while_loop_raises_compatibility_error_with_capture(self):
+        """Test that while_loop raises CompatibilityError when capture mode is enabled."""
+        qml.capture.enable()
+
+        def condition(i):
+            return i < 5
+
+        with pytest.raises(CompatibilityError) as exc_info:
+
+            @while_loop(condition)
+            def loop_fn(x):
+                return x + 1
+
+        # Verify the error message is specific and helpful
+        error_msg = str(exc_info.value)
+        assert "catalyst.while_loop is not supported with PennyLane's capture enabled" in error_msg
+
+    @pytest.mark.xfail(reason="autograph")
```

Comment:
What's going on here?

Reply:
So it will emit the error we just deisgned and be successfully caught. However, one more error from autograph will show up... I think it might be bad idea to introduce more dependency just for this test so xfailed it here

Reply:
> Is autograph always on with program capture? (I don't see it enabled in the test)
> 
> I wonder what is different about this test than the others. Can you share the error exactly?
> 
> I'd would like to avoid just having an xfail test for no reason. What is the dependency you'd like to avoid?

I want to avoid importing and catching the `AutoGraphError` here

Reply:
Oh never mind, I just read the error msg and replacing lambda resolved it 😓

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1945/commits/6db25ece6d732c99e51a5b4f16a7c41d16ba422a

Reply:
> This is news to me 🤔

I think it's a known problem with AG, or least at some point it was. But yeah different problem!